example/provider-extensions/garden/configure-admission.sh

```diff
@@ -0,0 +1,63 @@
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses~LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+
+SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )"
+REPO_ROOT_DIR="$(realpath "$SCRIPT_DIR"/../../..)"
+
+usage() {
+  echo "Usage:"
+  echo "> configure-admission.sh [ -h | <garden-kubeconfig> <apply|delete> ]"
+  echo
+  echo ">> For example: configure-admission.sh ~/.kube/garden-kubeconfig.yaml apply"
+
+  exit 0
+}
+
+if [ "$1" == "-h" ] || [ "$#" -ne 2 ]; then
+  usage
+fi
+
+if [ "$2" != "apply" ] && [ "$2" != "delete" ]; then
+  usage
+fi
+
+garden_kubeconfig=$1
+command=$2
+
+for p in $(yq '. | select(.kind == "ControllerDeployment") | select(.metadata.name == "provider-*") | .metadata.name' $REPO_ROOT_DIR/example/provider-extensions/garden/controllerregistrations/* | grep -v -E "^---$"); do
+  echo "Found \"$p\" in $REPO_ROOT_DIR/example/provider-extensions/garden/controllerregistrations. Trying to configure its admission controller..."
+  if PROVIDER_RELEASES=$(curl --fail -s -L -H "Accept: application/vnd.github+json" https://api.github.com/repos/gardener/gardener-extension-$p/releases); then
+    LATEST_RELEASE=$(jq -r '.[].tag_name' <<< $PROVIDER_RELEASES | head -n 1)
+    ADMISSION_NAME=${p/provider/admission}
+    echo "Identified $LATEST_RELEASE as latest release of $ADMISSION_NAME. Trying to deploy it..."
+    ADMISSION_GIT_ROOT=$(mktemp -d)
+    git clone https://github.com/gardener/gardener-extension-$p $ADMISSION_GIT_ROOT
+    git -C $ADMISSION_GIT_ROOT checkout $LATEST_RELEASE
+    CA_BUNDLE=$(yq ".webhooks[0].clientConfig.caBundle" $ADMISSION_GIT_ROOT/example/50-mutatingwebhookconfiguration.yaml | base64 -d)
+    TLS_CRT=$(cat $ADMISSION_GIT_ROOT/example/$ADMISSION_NAME-certs/tls.crt)
+    TLS_KEY=$(cat $ADMISSION_GIT_ROOT/example/$ADMISSION_NAME-certs/tls.key)
+    helm template --namespace garden --set global.webhookConfig.caBundle="$CA_BUNDLE" --set global.webhookConfig.tls.crt="$TLS_CRT" --set global.webhookConfig.tls.key="$TLS_KEY" gardener-extension-$ADMISSION_NAME $ADMISSION_GIT_ROOT/charts/gardener-extension-$ADMISSION_NAME/charts/application > $ADMISSION_GIT_ROOT/virtual-resources.yaml
+    helm template --namespace garden --set global.webhookConfig.caBundle="$CA_BUNDLE" --set global.webhookConfig.tls.crt="$TLS_CRT" --set global.webhookConfig.tls.key="$TLS_KEY" --set global.kubeconfig="$(cat $garden_kubeconfig | sed 's/127.0.0.1:.*$/kubernetes.default.svc.cluster.local/g')" --set global.vpa.enabled="false" gardener-extension-$ADMISSION_NAME $ADMISSION_GIT_ROOT/charts/gardener-extension-$ADMISSION_NAME/charts/runtime > $ADMISSION_GIT_ROOT/runtime-resources.yaml
```

Comment:
```suggestion
    helm template --namespace garden --set global.image.tag="$LATEST_RELEASE" --set global.webhookConfig.caBundle="$CA_BUNDLE" --set global.webhookConfig.tls.crt="$TLS_CRT" --set global.webhookConfig.tls.key="$TLS_KEY" --set global.kubeconfig="$(cat "$garden_kubeconfig" | sed 's/127.0.0.1:.*$/kubernetes.default.svc.cluster.local/g')" --set global.vpa.enabled="false" gardener-extension-"$ADMISSION_NAME" "$ADMISSION_GIT_ROOT"/charts/gardener-extension-"$ADMISSION_NAME"/charts/runtime > "$ADMISSION_GIT_ROOT"/runtime-resources.yaml
```
In case we don't use the latest version, `global.image.tag` has to be overwritten too.

Reply:
Fair point. Adapted with https://github.com/gardener/gardener/pull/8311/commits/89402c058b31edd6af49bd0adbce94d7d2e22a3b.